### PR TITLE
Adds documentation for WordPress.WP.EnqueuedResources.

### PR DESCRIPTION
**Pull request by NielsdeBlaauw**
Related to #1722

### CODE REVIEW

File: WordPress/Docs/WP/EnqueuedResourcesStandard.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<documentation title="Enqueued Resources">
+    <standard>
+    <![CDATA[
+    Scripts must be registered/enqueued via wp_enqueue_script.
```
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourcesStandard.xml]:
```suggestion
    Scripts must be registered/enqueued via wp_enqueue_script().
```

File: WordPress/Docs/WP/EnqueuedResourcesStandard.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        ]]>
+        </code>
+        <code title="Invalid: Stylesheet is directly embedded in HTML.">
+        <![CDATA[
+<link rel="stylesheet" 
+      <em>href="http://someurl/somefile.css"</em> />
+        ]]>
+        </code>
+    </code_comparison>
+
```
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourcesStandard.xml]: Nitpick: (blank) line contains trailing whitespace.

File: WordPress/Docs/WP/EnqueuedResourcesStandard.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        <code title="Invalid: Script is directly embedded in HTML.">
+        <![CDATA[
+<script <em>src="http://someurl/somefile.js"</em>>
+</script>
+        ]]>
+        </code>
+    </code_comparison>
+    <standard>
+    <![CDATA[
+    Stylesheets must be registered/enqueued via wp_enqueue_style.
```
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourcesStandard.xml]:
```suggestion
    Stylesheets must be registered/enqueued via wp_enqueue_style().
```

File: WordPress/Docs/WP/EnqueuedResourcesStandard.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: Stylesheet registered and enqueued correctly.">
+        <![CDATA[
+<em>wp_enqueue_style</em>( 
+    'style-name', 
+    'https://example.com/somefile.css', 
+    false,
```
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourcesStandard.xml]: The third parameter of `wp_enqueue_style()` is expected to be an array.

Ref: https://developer.wordpress.org/reference/functions/wp_enqueue_style/

**Comment by NielsdeBlaauw** [comment]
> Aside from that, I'm wondering if it would be better for the example code to reference local files instead of external URLs, as that is (or should be) the more common use-case.

I'm worried the code for the invalid example might be too far from what the user is actually doing, or what the sniff will be picking up. It's not about what resource is loaded, it is more about the way it is being done. IMHO this example is easier and more accessible to people with limited knowledge of WP. Also it avoids a discussion of where the actual resource is coming from.

**Comment by jrfnl** [comment]
> > Aside from that, I'm wondering if it would be better for the example code to reference local files instead of external URLs, as that is (or should be) the more common use-case.
> 
> I'm worried the code for the invalid example might be too far from what the user is actually doing, or what the sniff will be picking up. It's not about what resource is loaded, it is more about the way it is being done. IMHO this example is easier and more accessible to people with limited knowledge of WP. Also it avoids a discussion of where the actual resource is coming from.

True enough. Maybe using a `$path_to_file` variable would simplify it in that case ?

The reason I'm not keen to leave the examples with the URLs is two-fold and revolves around other problems which this sniff is not sniffing for:
1. People using a CDN-version of script which WP natively ships with, often at a different version than the one which ships with WP.
    This breaks other plugins/themes as those expect the WP native version and their code may break when a different version is used. Also see #867 
2. People using just a CDN-version of a script without fall-back to local.
    This can break sites pretty badly when:
    - the CDN is down;
    - the CDN stops hosting a particular script or the particular version being pointed to;
    - the pointer is to a version-less script name and the version of the script on the CDN changes.

These are issues outside the scope of the sniff, so I agree that we shouldn't address them here, but I do believe showing code which we may well forbid at a later stage, is not a good idea either.

**Comment by NielsdeBlaauw** [comment]
Agree, I going to replace it with a variable.

File: WordPress/Docs/WP/EnqueuedResourcesStandard.xml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    '1.0.0',
+    true
+);
+        ]]>
+        </code>
+        <code title="Invalid: Script is directly embedded in HTML.">
+        <![CDATA[
+printf( 
+    '<em><script src="%s"></script></em>',
+    esc_attr( $path_to_file )
```
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourcesStandard.xml]:
```suggestion
    esc_url( $path_to_file )
```

File: WordPress/Docs/WP/EnqueuedResourcesStandard.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    array(), 
+    '1.0.0'
+);
+        ]]>
+        </code>
+        <code title="Invalid: Stylesheet is directly embedded in HTML.">
+        <![CDATA[
+printf( 
+    '<em><link rel="stylesheet" href="%s" /></em>',
+    esc_attr( $path_to_file )
```
Comment (jrfnl) [inline WordPress/Docs/WP/EnqueuedResourcesStandard.xml]:
```suggestion
    esc_url( $path_to_file )
```

**Pull request merged**